source/server-discovery-and-monitoring/server-monitoring.rst

```diff
@@ -682,27 +686,29 @@ The event API here is assumed to be like the standard `Python Event
         try:
             if not connection:
                 setUpConnection()
-                return new ServerDescription from handshake response
-
-            if connection.moreToCome:
-                read next isMaster exhaust response
-                return new ServerDescription
-
-            if previousDescription.topologyVersion:
+                rttMonitor.addSample(connection.handshakeDuration)
+                response = connection.handshakeResponse
+            elif connection.moreToCome:
+                response = read next isMaster exhaust response
+            elif previousDescription.topologyVersion:
                 # Initiate streaming isMaster
+                rttMonitor.start()
```

Comment:
nit: In the changes for the RTT thread description below, this method is called `run`.

Reply:
Fixed.